presto-main/src/main/java/com/facebook/presto/operator/project/InterpretedCursorProcessor.java

```diff
@@ -87,7 +88,7 @@ private static ExpressionInterpreter getExpressionInterpreter(
             parameterTypes.put(parameter, type);
         }
 
-        IdentityLinkedHashMap<Expression, Type> expressionTypes = getExpressionTypesFromInput(session, metadata, sqlParser, parameterTypes.build(), rewritten, emptyList());
+        IdentityLinkedHashMap<Expression, Type> expressionTypes = NodeRefCollections.toIdentityMap(getExpressionTypesFromInput(session, metadata, sqlParser, parameterTypes.build(), rewritten, emptyList()));
```

Comment:
I don't understand? You mean why i use `NodeRef<Expression>` instead of (more correct) `NodeRef<? extends Expresssion>`? If so, reason is legibility of methods signatures.